new_is_always_better.py

```diff
@@ -0,0 +1,96 @@
+from scapy.all import conf
+import argparse
+from struct import unpack
+
+
+BROADCAST = b'\xff\xff\xff\xff\xff\xff'
+DST_MAC_INDEX = 0
+SRC_MAC_INDEX = 1
+TYPE_INDEX = 2
+PAYLOAD_INDEX = 3
+
+
+def int_to_hex(num: int) -> hex:
+    """
+    Converts an integer to a hex representation.
+    :param num: the number to convert.
+    :return: the hex representation of the number.
+    """
+    return hex(num)
+
+
+def to_send(fields: tuple) -> bytes:
+    """
+    Gets the part of the frame to send to the next layer.
+    :param fields: the fields of the processed frame received.
+    :return: the part of the frame to send to the next layer.
+    """
+    bytes_to_send = fields[TYPE_INDEX] + fields[PAYLOAD_INDEX]
+    return bytes_to_send
+
+
+def is_multicast(recv: tuple) -> bool:
+    """
+    Checks if the address returned is a multicast address.
+    :param recv: the message received.
+    :return: True if it is a multicast address, False otherwise.
+    """
+    if int(recv[1][0]) % 2 == 1:
+        return True
+    return False
+
+
+def unpack_frame(recv: tuple) -> tuple:
+    """
+    Unpacks the frame received.
+    :param recv: the frame received.
+    :return: a tuple of the different parts of the frame.
+    """
+    frame_size = len(recv[1])
+    return unpack(f"6s6s2s{frame_size - 14}s", recv[1])
+
+
+def process_frame(iface: str, iface_mac: str) -> bytes:
+    """
+    Processes the frame - returns it if it belongs to the right interface, or drops it otherwise.
+    :return: the packet to send to the next layer, or b'' if the packet was sent to a different mac address.
+    """
+    for_me = True
+    sock = conf.L2socket(iface=iface, promisc=True)
+    recv = None
+
+    # sniffing until a frame is received
+    while recv is None or recv == (None, None, None):
+        recv = sock.recv_raw()
+    fields = unpack_frame(recv)
```

Comment:
It is true that using magic numbers is usually discouraged, but you shouldn't have to write a ton of new constants (DST_MAC_INDEX, SRC_MAC_INDEX....) whenever you have a function with multiple return values. 

If you prefer to avoid writing unexplained numbers but want to skip the constants, a nicer way to do this might be unpacking:

```suggestion
    dst_mac, src_mac, type, payload = unpack_frame(recv)
```

Reply:
fixed